jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocInfo.java

```diff
@@ -0,0 +1,178 @@
+        if (params.otherSessionConfs() != null) {
+            hiveJdbcURL = String.format("%s;%s", hiveJdbcURL, params.otherSessionConfs());
+        }
+        if (params.hiveConfs() != null) {
+            hiveJdbcURL = String.format("%s?%s", hiveJdbcURL, params.hiveConfs());
+        }
+        if (params.hiveVars() != null) {
+            hiveJdbcURL = String.format("%s#%s", hiveJdbcURL, params.hiveVars());
+        }
+        return hiveJdbcURL;
+    }
+
+    /**
+     * Use Dataproc client libraries to get cluster.
+     *
+     * @return name of the cluster's master node
+     * @throws InvalidURLException
+     */
+    public String getHost() throws InvalidURLException {
+        String host =
+                params.clusterName() == null
+                        ? findClusterInPool(formatClusterFilterString())
+                        : getClusterByName();
+        return host + "-m";
+    }
+
+    /**
+     * Supports finding & verifying the cluster indicated by name.
+     *
+     * @return the cluster of the given name
+     * @throws InvalidURLException
+     */
+    public String getClusterByName() throws InvalidURLException {
+        try {
+            Cluster cluster =
+                    clusterControllerClient.getCluster(
+                            params.projectId(), params.region(), params.clusterName());
+            return cluster.getClusterName();
+        } catch (ApiException e) {
+            throw new InvalidURLException(
+                    String.format(
+                            "Unable to retrieve cluster information for %s/%s/%s.\n",
+                            params.projectId(), params.region(), params.clusterName()),
+                    e);
+        }
+    }
+
+    /**
+     * Formats the user passed in clusterPoolLabel value to the filter format that Dataproc API
+     * accpets: field = value [AND [field = value]] ...
+     *
+     * @return formatted filter string accepted by Dataproc Client API
+     */
+    public String formatClusterFilterString() throws InvalidURLException {
+        Map<String, List<String>> labels = new LinkedHashMap<>();
+        // `ACTIVE` contains the `CREATING`, `UPDATING`, and `RUNNING` states
+        labels.put("status.state", ImmutableList.of("ACTIVE"));
+
+        // clusterPoolLabel can be null, but filter is not empty string since
+        // it at least has "status.state = ACTIVE"
+        if (params.clusterPoolLabel() != null) {
+            for (String label : params.clusterPoolLabel().split(":")) {
+                checkUrl(
+                        !label.isEmpty() && label.contains("="),
+                        "'%s' Invalid clusterPoolLabel.\n"
+                                + "Example format: clusterPoolLabel=field1=value1:field2=value2'",
+                        label);
+                String field = label.split("=")[0];
+                String value = label.split("=")[1];
+
+                if (field.startsWith("status")) {
```

Comment:
Nit: as discussed offline use `equals("status.state")` to ensure we don't block user labels